components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/api/UsernameRecoveryApi.java

```diff
@@ -233,4 +224,17 @@ public void recoverUsernamePost(List<UserClaim> claim, String tenantDomain, Bool
     return apiClient.invokeAPI(localVarPath, "GET", localVarQueryParams, null, localVarHeaderParams, localVarFormParams,
             localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
   }
+
+  private void setBasePath(boolean isEndpointTenantAware, String tenantDomain) {
+
+    if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+      basePath = IdentityManagementEndpointUtil
+              .buildTenantQualifiedUrl(IdentityManagementEndpointConstants.UserInfoRecovery.USER_API_RELATIVE_PATH);
+    } else if (isEndpointTenantAware && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equalsIgnoreCase(tenantDomain)) {
```

Comment:
What is this 'isEndpointTenantAware' property is used for ?

Reply:
It's has been used in legacy mode to defend whether the endpoint needs to be tenant qualified.